libbeat/outputs/elasticsearch/client.go

```diff
@@ -251,6 +251,20 @@ func (client *Client) publishEvents(
 	return nil, nil
 }
 
+func eventMetaValue(event *beat.Event, key string) string {
+	var val string
+	if m := event.Meta; m != nil {
+		if tmp := m[key]; tmp != nil {
+			if s, ok := tmp.(string); ok {
+				val = s
+			} else {
+				logp.Err("Event[%s] '%v' is no string value", key, val)
```

Comment:
I believe what you want here is `tmp` instead of an empty `val`. 

i.e.
```suggestion
				logp.Err("Event[%s] '%v' is no string value", key, tmp)
```

This is also wrong in current code